framework/src/main/java/org/apache/drill/test/framework/TestDriver.java

```diff
@@ -276,14 +326,25 @@ public int runTests() throws Exception {
         LOG.info(test.getInputFile());
         LOG.info("Query: \n" + test.getQuery());
       }
-      LOG.info("Random Failures:");
+      if(randomFailures.size()>0){
+        LOG.info("Random Failures:");
+      }
       for (DrillTest test : randomFailures) {
         LOG.info(test.getInputFile());
         LOG.info("Query: \n" + test.getQuery());
         LOG.info("Failed with exception", test.getException());
       }
+      if(skippedTests.size() > 0) {
+        LOG.info("Skipped Tests: These tests cannot be run from " + queryLogFilename + " because they were not in source or group");
+        for (String line : skippedTests) {
+          LOG.info(line);
+        }
+      }
       LOG.info(LINE_BREAK);
       LOG.info("Summary");
+      if ((cmdParam.repeatRun == true) && (queryFile.exists() && (queryFile.length() > 0))) {
```

Comment:
Can queryFile.length change? For example, if you started without the file but created afterwards, will this evaluate to true during the same run?